src/components/button/config/demo.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright IBM Corp. 2018, 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * Configuration for the Button component
+ * @module button/demo
+ */
+import startCase from 'lodash/startCase';
+import { generate } from './generate';
+
+/**
+ * Creates a button template configuration for a variant
+ * @returns {globalTypedefs.fractalDemo} fractal demo object
+ */
+const variantButtons = function() {
+  const variants = {};
+
+  Object.keys(this.selectors.variants).forEach(variant => {
+    variants[variant] = {
+      name: variant,
+      label: startCase(variant),
+      context: generate.apply(this, [
+        {
+          variant,
+        },
+      ]),
+    };
+  });
+
+  return variants;
+};
+
+/**
+ * Modifier: Anchor
+ * @returns {globalTypedefs.fractalDemo} fractal demo object
+ */
+const anchorButton = function() {
+  return {
+    name: 'anchor',
+    label: 'Anchor',
+    context: generate.apply(this, [
+      {
+        element: 'a',
+      },
+    ]),
+  };
+};
+
+/**
+ * Modifier: Small
+ * @returns {globalTypedefs.fractalDemo} fractal demo object
+ */
+const smallButton = function() {
+  return {
+    name: 'small',
+    label: 'Small',
+    context: generate.apply(this, [
```

Comment:
* putting the `context` which is used by the `generate` function within the `generate.apply` call means that that object is not available for use by the frameworks, all they would receive is the post-generate object. We put this in a separate `preGenerate` variable, which allows both use of the smaller config object directly...as well as giving us access to the preGenerated object in documentation
* why is this within an array?

Reply:
1. Have you found use for that pre-generate object outside of documentation? I'll update.
1. Second parameter of `function.apply()` is of type array.